### PR TITLE
fix(cache): run cache init

### PR DESCRIPTION
**Pull request by rarkins**
Refactors the in-memory run cache use so that it is inactive in tests unless init() called first.

### CODE REVIEW

File: lib/datasource/maven/index.ts [hunk 1]
```diff
@@ -178,6 +178,7 @@ async function getVersionsFromMetadata(
 
 type ArtifactsInfo = Record<string, boolean | null>;
 
+// istanbul ignore next
```
Comment (viceice) [inline lib/datasource/maven/index.ts:181]: We should add a test for this later

File: lib/util/cache/run.ts [hunk 2]
```diff
@@ -1,13 +1,15 @@
-let repoCache: Record<string, any> = {};
+let repoCache: Record<string, any>;
 
-export function clear(): void {
+export function init(): void {
   repoCache = {};
 }
```
Comment (viceice) [inline lib/util/cache/run.ts:5]:
```suggestion
export function init(): void {
  repoCache = {};
}
export function reset(): void {
  repoCache = undefined;
}
```

Function to unset cache for tests

File: lib/util/cache/run.ts [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
 
 export function get<T = any>(key: string): T {
-  return repoCache[key];
+  return repoCache ? repoCache[key] : undefined;
```
Comment (viceice) [inline lib/util/cache/run.ts:8]:
```suggestion
  return repoCache.?[key];
```

File: lib/util/cache/run.ts [hunk 4]
```diff
@@ -1,13 +1,15 @@
-let repoCache: Record<string, any> = {};
+let repoCache: Record<string, any>;
```
Comment (viceice) [inline lib/util/cache/run.ts:1]:
```suggestion
let repoCache: Record<string, any> | undefined;
```

**Review by viceice — COMMENTED** [review]
Just some small suggestions

File: lib/datasource/maven/index.ts (hunk 1 above)
Comment (rarkins) [reply lib/datasource/maven/index.ts:181]: Yeah, I'm not sure why it lost coverage but seems the previous test depended on the cache being set

**Pull request merged**